### PR TITLE
feat(carousel): add onSlide callback

### PR DESCRIPTION
**Pull request by paghar**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change contains documentation update

## Breaking changes

Please document the breaking changes if suitable.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -108,6 +110,10 @@ export const Carousel: FC<CarouselProps> = ({
 
   const handleDragging = (dragging: boolean) => () => setIsDragging(dragging);
 
+  useEffect(() => {
+    !!onSlide && onSlide(activeItem);
```
Comment (rluders) [inline src/components/Carousel/Carousel.tsx:114]: Why do you add two `!!`? Isn't just `onSlide` enough?

**Review by rluders — CHANGES_REQUESTED** [review]

File: src/components/Carousel/Carousel.tsx (hunk 1 above)
Comment (paghar) [reply src/components/Carousel/Carousel.tsx:114]: @rluders !! is not necessary and i just added it for readability sake, i can remove it if you prefer.

Comment (rluders) [reply src/components/Carousel/Carousel.tsx:114]: @paghar yes, please make it simplier, just `onSlide && onSlide(activeItem)` is good.

**Comment by rluders** [comment]
Replaced by #833

**Pull request closed without merging**